frontend/public/components/routes/create-route.tsx

```diff
@@ -426,7 +430,8 @@ class CreateRouteWithTranslation extends React.Component<
           {!_.isEmpty(portOptions) && (
             <Dropdown
               items={portOptions}
-              title={portOptions[targetPort] || t('public~Select target port')}
+              title={t('public~Select target port')}
+              selectedKey={_.isNumber(targetPort) ? UNNAMED_PORT_KEY : targetPort} // if targetPort is a number, it's an unnamed port
```

Comment:
Fix so the `Target port` dropdown is correctly populated when editing (before the dropdown did not have a selected value).